internal/printer/print-to-tsx.go

```diff
@@ -64,6 +64,10 @@ func getTextType(n *astro.Node) TextType {
 func renderTsx(p *printer, n *Node) {
 	// Root of the document, print all children
 	if n.Type == DocumentNode {
+		props := js_scanner.GetPropsType([]byte(p.sourcetext))
+		if props.Ident != "Record<string, any>" {
+			p.printf(`declare const Astro: { props: %s } & Omit<import('astro').AstroGlobal, 'props'>`, props.Ident)
```

Comment:
It's a super unfortunate behavior, but the JSDoc comment for both the Astro const and the props attributes needs to be included or they get lost

What I'm thinking we could do on the Astro side is a type for the Astro global that takes a generic to type the props, that way we wouldn't need the Omit or to include the JSDoc, thoughts?

Reply:
Yep, that makes a lot of sense to me!

Reply:
Made a PR for it: https://github.com/withastro/astro/pull/5000